src/Microsoft.ML.Transforms/MissingValueDroppingTransformer.cs

```diff
@@ -229,110 +229,110 @@ private void DropNAsAndDefaults<TDst>(ref VBuffer<TDst> src, ref VBuffer<TDst> d
         {
             Host.AssertValue(isNA);
 
+            var srcValues = src.GetValues();
             int newCount = 0;
-            for (int i = 0; i < src.Count; i++)
+            for (int i = 0; i < srcValues.Length; i++)
             {
-                if (!isNA(in src.Values[i]))
+                if (!isNA(in srcValues[i]))
                     newCount++;
             }
-            Host.Assert(newCount <= src.Count);
+            Host.Assert(newCount <= srcValues.Length);
 
             if (newCount == 0)
             {
-                dst = new VBuffer<TDst>(0, dst.Values, dst.Indices);
+                dst = VBufferMutationContext.Create(ref dst, 0)
+                    .CreateBuffer();
```

Comment:
`Create` followed immediately by `CreateBuffer` seems to be a common pattern too. Maybe add `VBufferMutationContext.Resize` for that?

Reply:
That method seems like it would belong in `VBufferUtils`. I will add it there.

---
In reply to: [233679182](https://github.com/dotnet/machinelearning/pull/1580#discussion_r233679182) [](ancestors = 233679182)